spec/moonshot/plugins/rotate_asg_instances/asg_spec.rb

```diff
@@ -262,4 +262,30 @@ def stub_cf_client
       end
     end
   end
+
+  describe '#instance_in_terminal_state?' do
+    context 'when an ec2 instance object is nil' do
+      let(:terminated_ec2_instance) do
+        Aws::EC2::Instance.new(id: "i-123456")
+      end
+      let(:check_terminal_state){ subject.send(:instance_in_terminal_state?, terminated_ec2_instance)}
+      it 'should send true for  state for nil instance object.' do
+        allow(terminated_ec2_instance).to receive(:exists?).and_return(false)
+        expect(check_terminal_state).to match(true)
```

Comment:
It's best practice to have `allow` outside the `it` block. You can put it in before block.

Reply:
Updated